src/phpDocumentor/Descriptor/Collection.php

```diff
@@ -27,18 +27,19 @@
  * The goal for this class is to allow Descriptors to be easily retrieved and set so that interaction in
  * templates becomes easier.
  *
- * @template-implements ArrayAccess<string|int, mixed>
- * @template-implements IteratorAggregate<string|int, mixed>
+ * @template T
```

Comment:
A generic! How wonderful and distressing at the same time! :D Not because generics are bad; but because phpDocumentor does not support them yet :sweat_smile: Sounds like a good 'eat your own dogfood' case to build support for them :) (We do want this in 3.1)

Reply:
Yeah, really useful to use, but it must have been a pain to implement!

Reply:
They are a completely new type of syntax that breaks with the way tags were parsed. In the past, you could split a tag based on whitespace to get its parts; but with things like generics you can no longer do that and need a more intelligent parser

Reply:
I believe PHPStan parser is a distinct repo on github and it can be reused. IIRC, they used it in slevomat to read annotations and fix the issue where it believed an import was not use (which I suppressed)

Reply:
We have our own component that supports quite a few more features than other DocBlock parsers out there; but not (yet) generics :D

Reply:
It would actually have been nice if other projects would contribute a missing feature instead of building another DocBlock parser :sweat_smile:

Reply:
True. I don't know why it was done that way :(